### ISSUE TITLE
Parser does not error when there are duplicate keys

### ISSUE THREAD

**Issue by stephenmac7** [kowainik/tomland#211]
The TOML `"x = 1\nx = 2"` is parsed as `"x = 2"`, but it should error according to the TOML spec.

This is actually a pretty large issue because misspellings interact badly with this behavior. For example,
```
[[item]]
  [[item.sub]]
    a = 1
[[item]]
  [[item.sub]]
    a = 2
  [[ite.sub]]
    a=4
[[item]]
  [[item.sub]]
    a = 3
```
parses as
```
[[ite.sub]]
  a = 4

[[item]]
  [[item.sub]]
    a = 3
```

which is quite different from what we would like.

**Comment by jiegillet** [comment]
Yes, this is basically #5, we haven't implemented any checks yet. If @chshersh is OK with it, I'm happy to get started on that. 

The examples you give are very relevant, thank you.

**Comment by chshersh** [comment]
@stephenmac7 Thanks for providing a convincing example for this feature! As @jiegillet mentioned, this is indeed a part of TOML validation, which hasn't been implemented yet. Parsing TOML (as far as `tomland` concerned) is a multi-step process which includes lexing, parsing, validation and decoding. I see that @jiegillet already opened a PR. Will look into this soon.

**Comment by chshersh** [comment]
I'm currently slowly working on refactoring the parser and implementing validation. But I want to point out that _validation_ is not the only solution to this particular problem. The alternative solution is to implement TOML difference (#186). So you can have warnings of unused fields. And the result of parsing TOML will be a data type like this:

```haskell
data Result a
    = Failure String
    | Warning [String] a
    | Success a
```

**Comment by chshersh** [comment]
@stephenmac7 I've implemented validation for duplicate keys in #216. However, I realised that your problem is not in duplicating keys. And checking for duplicating keys won't help you to solve your problem. What you need is TOML difference implementation. So I'm closing this issue as a duplicate of #186.
